### PR TITLE
Add post object field format arg

### PR DESCRIPTION
**Pull request by chriszarate**
_Edited to reflect discussion and changes:_

This provides a method to request `RAW` or `RENDERED` format of post object fields:

```
query {
  post(id: "...") {
    content(format: RAW)
  }
}
```

The `format` arg is available on `content`, `title`, and `excerpt`. If the `format` arg is omitted, then the default behavior will be `RENDERED` (which is simply to apply these filters: `the_content` on `content`, `the_title` on `title`, and `get_the_excerpt` and `the_excerpt` on `excerpt`).

It introduces a new enum type, `PostObjectFieldFormatEnumType`.

Perhaps more significantly, it sets up and resets post data during post object field resolution, so that any custom resolution undertaken by the user will have that done for them.

### CODE REVIEW

File: src/Type/Enum/PostObjectFieldFilterEnumType.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?php
```
Comment (jasonbahl) [inline src/Type/Enum/PostObjectFieldFilterEnumType.php]: This file name needs to change to: PostObjectFieldFormatEnumType

Comment (jasonbahl) [reply src/Type/Enum/PostObjectFieldFilterEnumType.php]: Not sure how unit tests didn't bomb on this 🤔

Comment (chriszarate) [reply src/Type/Enum/PostObjectFieldFilterEnumType.php]: !!! ... is composer saving the day here?

**Comment by jasonbahl** [comment]
@chriszarate I added a few more tests, and all looks good. Merging! 

Thanks for your work on this!

**Comment by bq14** [comment]
@jasonbahl @chriszarate 

Testing this with
```
curl 7.76.1
PHP 8.0.7
WP GraphQL v1.4.3
WP v5.7.2
```

for case: format = RAW, queries return 'null'

I'm not certain I didn't misread and maybe it's my usage :-/


for default query,
```
$CURL \
  --data '{ "query": "{ pages(where: {title: \"Test1\"}) { edges { node { content } } } }" }' \
https://wptest.loc/cms/graphql/ | json_pp


	{
	   "data" : {
	      "pages" : {
	         "edges" : [
	            {
	               "node" : {
	                  "content" : "\n<p>test 123 content </p>\n\n\n\n<p>test 456 content</p>\n"
	               }
	            }
	         ]
	      }
	   },
	   "extensions" : {
	      "debug" : []
	   }
	}
```

format = RENDERED, same return as default,
```
$CURL \
  --data '{ "query": "{ pages(where: {title: \"Test1\"}) { edges { node { content(format: RENDERED) } } } }" }' \
https://wptest.loc/cms/graphql/ | json_pp

	{
	   "data" : {
	      "pages" : {
	         "edges" : [
	            {
	               "node" : {
	                  "content" : "\n<p>test 123 content </p>\n\n\n\n<p>test 456 content</p>\n"
	               }
	            }
	         ]
	      }
	   },
	   "extensions" : {
	      "debug" : []
	   }
	}
```

format = RAW, returns 'null'
```
$CURL \
  --data '{ "query": "{ pages(where: {title: \"Test1\"}) { edges { node { content(format: RAW) } } } }" }' \
https://wptest.loc/cms/graphql/ | json_pp

	{
	   "data" : {
	      "pages" : {
	         "edges" : [
	            {
	               "node" : {
	                  "content" : null
	               }
	            }
	         ]
	      }
	   },
	   "extensions" : {
	      "debug" : []
	   }
	}
```

**Comment by chriszarate** [comment]
> for case: format = RAW, queries return 'null'

@bq14 WPGraphQL's model layer prevents unauthorized access to certain data, following the protections of WordPress core. Raw post content is only exposed to users with `edit_posts` capability. Your example request is unauthenticated, therefore access to raw post content is not allowed:

https://github.com/wp-graphql/wp-graphql/blob/develop/src/Model/Post.php#L495

This PR predates the model layer, so apologies that there was no reference to this implementation detail.

If you want to expose raw post content, you'll need to authenticate your request or override the resolver for the `content` field to bypass this protection (be sure to evaluate the risks of doing so).

**Pull request merged**